### DISCUSSION TITLE
Window Resizing and Browser APIs

### DISCUSSION THREAD

**Discussion by raguay** [wailsapp/wails#810] [Q&A] [answered]
I'm moving a project from NW.js to Wails. I was using Tauri, but the websockets only work with a secure server even on a local host. But, I was able to get it to work okay in Wails. Now, I want to dynamically change window size based on the main container size. But, I don't see any API for changing window size. Is this possible?

Also, it seems the ResizeObserver API isn't supported. What browser APIs are supported? Is there a list somewhere?

Also, how do you do a tray icon? I think xBar is using Wails and has a tray icon, but I don't see anything in the docs. Will I need to use a separate go library?

Thanks for your help.

**Comment by raguay** [comment]
Okay, I figured out how to resize with `window.resizeTo(400,400)` and to move it with `window.moveTo(0,0)`. Now, if I could just remove the title bar. How do you do that?

**Reply by leaanthony** [reply]
Titlebar, you can change the type of titlebar on Mac if you're using the `v2-alpha-xbar` branch by using the "Titlebar" key in the "Mac" options (Check out `pkg/options`).
Regarding the tray, the best docs you'll have at this point is to look at the xbar source. You basically add a `png` to `build/tray` and use a runtime method (check the xbar source) to set it. Sorry I can't be more help right now but the focus is getting v2 out.

**Comment by raguay** [comment]
I also found out that V2 has what I need for menu tray icons and such. Should I go ahead and build with v1 and then migrate to V2 when it's at least beta quality, or is it better to just jump in and make it V2. I'm currently only using straight JS/HTML/CSS and the minimal go for launching it (it still had the hello example). I'm just now getting into doing Wails specific items (refactoring has been taking me a while).

**Comment by leaanthony** [comment, marked as answer]
Hey there! Thanks for your messages. V1 is using a fairly old webview so the options may be limited in terms of API support. The v2 situation is....interesting. Xbar does indeed use a v2 version and does indeed support menu trays and icons so you are welcome to build against that branch (`v2-alpha-xbar`) if it satisfies your needs. Be aware though that to ensure v2 has parity across the 3 platforms, it is unlikely menu trays will be supported, even on Mac for the v2 release. However, that branch will stay and of course that may be fine for you. The intent *is* to add tray menus after v2, which for Mac will be easy, but that will be a post v2 option. As a side note, the *way* v2 works has changed since the xbar branch: it now does not require any frontend bundling or compiling assets at all - it just accepts an `embed.FS`. I have started to port the Mac code (95% of which already exists in the xbar branch) and expect the Mac beta out ~1 month after the Windows beta. 
I hope that makes sense. If not, feel free to raise questions.

**Comment by raguay** [comment]
Thanks. How do you migrate to v2-alpha-xbar? Do I have to first remove the v1 binaries and then compile v2-alpha-xbar and recreate the directory? And the move over my current code?

**Reply by leaanthony** [reply]
Yeah that's a good question. I don't have good docs for that, but the main difference is the main.go file. Check that out.
